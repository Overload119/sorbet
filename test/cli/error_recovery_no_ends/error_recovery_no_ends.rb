# typed: true

class A
  def foo(x)
    if x
      puts 'hello'
