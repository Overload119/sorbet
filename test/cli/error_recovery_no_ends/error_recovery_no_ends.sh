#!/usr/bin/env bash

set -euo pipefail

stderr_log="$(mktemp)"
trap 'rm -rf "$stderr_log"' EXIT

if main/sorbet \
    --silence-dev-message \
    --print=index-tree \
    test/cli/error_recovery_no_ends/error_recovery_no_ends.rb 2> "$stderr_log"; then
  echo "Expected to fail!"
  exit 1
fi

echo -------------------------------------

cat "$stderr_log"
