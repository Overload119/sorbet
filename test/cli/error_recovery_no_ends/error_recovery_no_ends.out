class <emptyTree><<C <root>>> < (::<todo sym>)
  class <emptyTree>::<C A><<C <todo sym>>> < (::<todo sym>)
    def foo<<todo method>>(x, &<blk>)
      if x
        <self>.puts("hello")
      else
        <emptyTree>
      end
    end

    ::Sorbet::Private::Static.keep_def(<self>, :foo, :normal)
  end
end
-------------------------------------
test/cli/error_recovery_no_ends/error_recovery_no_ends.rb:6: unexpected token "end of file" https://srb.help/2001
     6 |      puts 'hello'
     7 |

test/cli/error_recovery_no_ends/error_recovery_no_ends.rb:6: unexpected token "end of file" https://srb.help/2001
     6 |      puts 'hello'
     7 |

test/cli/error_recovery_no_ends/error_recovery_no_ends.rb:6: unexpected token "end of file" https://srb.help/2001
     6 |      puts 'hello'
     7 |
Errors: 3
